eng/scripts/Attest-CPEX-KPIs.ps1

```diff
@@ -0,0 +1,162 @@
+    $lifecycleToMgmtKpis = @{
+        "In Dev" = @($KPI_ID_Mgmt_Private_Preview, $KPI_ID_Mgmt_Public_Preview, $KPI_ID_Mgmt_GA)
+        "Private Preview" = @($KPI_ID_Mgmt_Public_Preview, $KPI_ID_Mgmt_GA)
+        "Public Preview" = @($KPI_ID_Mgmt_GA)
+    }
+
+    if ($mgmtAttestationStatus -ne "Completed") {
+        if ($mgmtScope -eq "Yes") {
+            Update-AttestationStatusInWorkItem -workItemId $triage.id -fieldName "Custom.ManagementPlaneAttestationStatus" -status "Completed"
+        } else {
+            Update-AttestationStatusInWorkItem -workItemId $triage.id -fieldName "Custom.ManagementPlaneAttestationStatus" -status "Not applicable"
+            if ($lifecycleToMgmtKpis.ContainsKey($productLifecycle)) {
+                foreach ($kpiId in $lifecycleToMgmtKpis[$productLifecycle]) {
+                    AddAttestationEntry $productServiceTreeId $kpiId $NA $productType $url
+                }
+            }
+        }
+    }
+
+}
+
+$releasePlans = Get-ReleasePlansForCPEXAttestation -releasePlanWorkItemId $ReleasePlanWorkItemId
+
+foreach ($releasePlan in $releasePlans) {
+    $fields = $releasePlan.fields
+    $dataScope = $fields["Custom.DataScope"]
+    $mgmtScope = $fields["Custom.MgmtScope"]
+    $lifecycle = $fields["Custom.ReleasePlanType"]
+    $productServiceTreeId = $fields["Custom.ProductServiceTreeID"]
+    $productType = $fields["Custom.ProductType"]
+    $url = $releasePlan.url
+
+    if ($productType -eq "Sku") {
+        $productType = "ProductSku"
+    }
+    
+    if ($dataScope -eq 'Yes') {
+        switch -Wildcard ($lifecycle) {
+            "*Public Preview*" {
+                AddAttestationEntry $productServiceTreeId $KPI_ID_Data_Public_Preview $Completed $productType $url
+            }
+            "*Private Preview*" {
+                AddAttestationEntry $productServiceTreeId $KPI_ID_Data_Private_Preview $Completed $productType $url
+            }
+            "*GA*" {
+                AddAttestationEntry $productServiceTreeId $KPI_ID_Data_GA $Completed $productType $url
+            }
+            default {
+                Write-Output "Release Plan ID $($releasePlan.id): Dataplane in scope, unknown lifecycle $($lifecycle)"
+            }
+        }
+    }
+
+    if ($mgmtScope -eq 'Yes') {
+        switch -Wildcard ($lifecycle) {
+            "*Public Preview*" {
+                AddAttestationEntry $productServiceTreeId $KPI_ID_Mgmt_Public_Preview $Completed $productType $url
+            }
+            "*Private Preview*" {
+                AddAttestationEntry $productServiceTreeId $KPI_ID_Mgmt_Private_Preview $Completed $productType $url
+            }
+            "*GA*" {
+                AddAttestationEntry $productServiceTreeId $KPI_ID_Mgmt_GA $Completed $productType $url
+            }
+            default {
+                Write-Output "Release Plan ID $($releasePlan.id): Management plane in scope, unknown lifecycle $($lifecycle)"
+            }
+        }
+    }
```

Comment:
Only difference between these two blocks are KPI id for data v/s mgmt. You can set them to a kpi ID variables based on data or mgmt type and refer them to avoid duplicate code.